Controller/DbControllers/UserController.cs

```diff
@@ -129,6 +130,47 @@ public bool HasPermission(User user, Permissions permission, Permissions maxValu
             return max > maxValues[maxValue];
         }
 
+        /**
+         * Changes the details of a user
+         *
+         * @param user The user database object to insert
+         * @param newEmail New given email to check 
+         * @param newUsername New given username to check
+         *
+         * @return NewUserInfo, if it's valid the respective strings have been updated within the function.
+         */
+
+        public virtual NewUserInfo ChangeDetails(User user, string newEmail, string newUsername)
+        {
+            if (newEmail == "" && newUsername == "")
+                return NewUserInfo.Empty;
+            
+            var email = GetUserFromEmailOrUsername(newEmail);
+            var username = GetUserFromEmailOrUsername(newUsername);
+            var updatedEmail = false;
+
+            if (newEmail != "")
+            {
+                if (email != null)
+                    return NewUserInfo.EmailTaken;

```

Comment:
dus als email niet leeg is dan krijg je email taken terug?

Reply:
nee, als je email als bestaat in de database krijg je emailtaken, wordt in line 148 opgehaald

Reply:
aah ja, ik snap t